packages/core/src/controllers/ChainController.ts

```diff
@@ -187,6 +187,36 @@ export const ChainController = {
     )
   },
 
+  addNetwork(network: CaipNetwork) {
+    const chainAdapter = state.chains.get(network.chainNamespace)
+    if (chainAdapter) {
+      chainAdapter.caipNetworks = [...(chainAdapter.caipNetworks || []), network]
+      state.chains.set(network.chainNamespace, chainAdapter)
+      this.setRequestedCaipNetworks(chainAdapter.caipNetworks || [], network.chainNamespace)
+    }
+  },
+
+  removeNetwork(namespace: ChainNamespace, networkId: string | number) {
+    const chainAdapter = state.chains.get(namespace)
+    if (chainAdapter) {
+      // Check if network being removed is active network
+      const isActiveNetwork = state.activeCaipNetwork?.id === networkId
+
+      // Filter out the network being removed
+      chainAdapter.caipNetworks = chainAdapter.caipNetworks?.filter(
+        network => network.id !== networkId
+      )
+
+      // If active network was removed and there are other networks available, switch to first one
+      if (isActiveNetwork && chainAdapter?.caipNetworks?.[0]) {
+        this.setActiveCaipNetwork(chainAdapter.caipNetworks[0])
+      }
```

Comment:
what happens when there are no other networks? We should prevent removing the only network avail .